src/gui/transferlistdelegate.cpp

```diff
@@ -49,218 +51,242 @@
+TransferListDelegate::~TransferListDelegate()
+{
+}
 
-void TransferListDelegate::paint(QPainter * painter, const QStyleOptionViewItem & option, const QModelIndex & index) const {
-  QStyleOptionViewItemV2 opt = QItemDelegate::setOptions(index, option);
-  painter->save();
-  switch(index.column()) {
-  case TorrentModel::TR_AMOUNT_DOWNLOADED:
-  case TorrentModel::TR_AMOUNT_UPLOADED:
-  case TorrentModel::TR_AMOUNT_DOWNLOADED_SESSION:
-  case TorrentModel::TR_AMOUNT_UPLOADED_SESSION:
-  case TorrentModel::TR_AMOUNT_LEFT:
-  case TorrentModel::TR_COMPLETED:
-  case TorrentModel::TR_SIZE:
-  case TorrentModel::TR_TOTAL_SIZE: {
-      QItemDelegate::drawBackground(painter, opt, index);
-      opt.displayAlignment = Qt::AlignRight | Qt::AlignVCenter;
-      QItemDelegate::drawDisplay(painter, opt, option.rect, Utils::Misc::friendlyUnit(index.data().toLongLong()));
-      break;
-    }
-  case TorrentModel::TR_ETA: {
-      QItemDelegate::drawBackground(painter, opt, index);
-      opt.displayAlignment = Qt::AlignRight | Qt::AlignVCenter;
-      QItemDelegate::drawDisplay(painter, opt, option.rect, Utils::Misc::userFriendlyDuration(index.data().toLongLong()));
-      break;
-    }
-  case TorrentModel::TR_SEEDS:
-  case TorrentModel::TR_PEERS: {
-      QString display = QString::number(index.data().toLongLong());
-      qlonglong total = index.data(Qt::UserRole).toLongLong();
-      if (total > 0) {
-        // Scrape was successful, we have total values
-        display += " ("+QString::number(total)+")";
-      }
-      QItemDelegate::drawBackground(painter, opt, index);
-      opt.displayAlignment = Qt::AlignRight | Qt::AlignVCenter;
-      QItemDelegate::drawDisplay(painter, opt, opt.rect, display);
-      break;
-    }
-  case TorrentModel::TR_STATUS: {
-      const int state = index.data().toInt();
-      QString display;
-      switch(state) {
-      case BitTorrent::TorrentState::Downloading:
-        display = tr("Downloading");
-        break;
-      case BitTorrent::TorrentState::StalledDownloading:
-        display = tr("Stalled", "Torrent is waiting for download to begin");
-        break;
-      case BitTorrent::TorrentState::DownloadingMetadata:
-        display = tr("Downloading metadata", "used when loading a magnet link");
-        break;
-      case BitTorrent::TorrentState::ForcedDownloading:
-        display = tr("[F] Downloading", "used when the torrent is forced started. You probably shouldn't translate the F.");
-        break;
-      case BitTorrent::TorrentState::Allocating:
-        display = tr("Allocating", "qBittorrent is allocating the files on disk");
-        break;
-      case BitTorrent::TorrentState::Uploading:
-      case BitTorrent::TorrentState::StalledUploading:
-        display = tr("Seeding", "Torrent is complete and in upload-only mode");
+void TransferListDelegate::paint(QPainter * painter, const QStyleOptionViewItem & option, const QModelIndex & index) const
```

Comment:
Done. 
The status string is not located at its own function:
`QString TransferListDelegate::getStatusString(const int state) const`